app/jinja_filters.py

```diff
@@ -43,6 +43,32 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(dates):
+    """ This function format_cond_date accepts two dates, if a date is submitted that is used and is available,
+    if not the metadata date is made available The format (example) in the json is
+    {{[answers.date_from, exercise.start_date]|format_conditional_date}} """
+
+    if dates[0]:
+        datex = dates[0]
+    else:
+        datex = dates[1]
+
+    fmts = ('%Y', '%b %d, %Y', '%b %d, %Y', '%B %d, %Y', '%B %d %Y', '%m/%d/%Y', '%m/%d/%y', '%b %Y', '%B%Y',
+            '%b %d,%Y', '%d/%m/%Y', '%Y-%m-%d %H:%M:%S')
```

Comment:
As discussed, we should assume this method is called with only two date 'formats' - a `dd/mm/yyyy` string date from the answer store, or a Python Date object from the metadata. Then, for each date argument we can just test if it's already a Date object and if not convert it to one (assuming a string in the format of `dd/mm/yyyy`). 
The Date object can then be output as a date string in the appropriate format using the `format_date` method.